tests/files/gz-file.cpp

```diff
@@ -109,6 +104,55 @@ TEST_CASE("Write a gz file") {
     CHECK(file.readline() == "5467");
 }
 
+TEST_CASE("Append to a gz file") {
+    auto filename = NamedTempPath(".gz");
+
+    {
+        TextFile file(filename, File::APPEND, File::GZIP);
+        file.print("Append 1\n");
+        file.print("{}\n", 7645);
+    }
+
+    {
+        TextFile file(filename, File::APPEND, File::GZIP);
+        file.print("Append 2\n");
+        file.print("{}\n", 6754);
+    }
+
+    std::ifstream checking(filename, std::ios::binary);
+    REQUIRE(checking.is_open());
+    checking.seekg(0, std::ios::end);
+    auto size = static_cast<size_t>(checking.tellg());
+    checking.seekg(0, std::ios::beg);
+
+    auto content = std::vector<uint8_t>(size);
+    checking.read(reinterpret_cast<char*>(content.data()), static_cast<std::streamsize>(size));
+
+    // Byte 9 identify the OS in gzip files.
+    // Override it so we can check for the full file output
+    content[9] = 0xff;
+
+    auto expected = std::vector<uint8_t> {
+        0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x73,
+        0x2c, 0x28, 0x48, 0xcd, 0x4b, 0x51, 0x30, 0xe4, 0x32, 0x37, 0x33,
+        0x31, 0xe5, 0x02, 0x00, 0xf8, 0x06, 0xaf, 0x8d, 0x0e, 0x00, 0x00,
+        0x00, 0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03,
+        0x73, 0x2c, 0x28, 0x48, 0xcd, 0x4b, 0x51, 0x30, 0xe2, 0x32, 0x33,
+        0x37, 0x35, 0xe1, 0x02, 0x00, 0xc6, 0x09, 0x42, 0x21, 0x0e, 0x00,
+        0x00, 0x00,
+    };
```

Comment:
I am not really sure on how to get the test file contents, so I simply run the test to produce a test file and check this file with `gzip -l <temp_file>.gz`. Since `gzip` doesn't result in any errors, I assume that the file contents are valid and used them as hard coded reference...

Reply:
Do you mean checking what the file content does mean? In the write test it is done mostly as a regression test, the exact content is not that important.

Reply:
I was wondering if there is a systematic way to derive the bytes that are expected. Technically it is possible that my file is garbage and I'm comparing against a broken file. But this does not matter for a regression test and I assume that the file is not broken because gzip can read it. Just wanted to be sure.